aspnet/2-structured-data/Models/DbBookStore.cs

```diff
@@ -46,12 +46,13 @@ public void Delete(long id)
         // [START list]
         public BookList List(int pageSize, string nextPageToken)
         {
-            var pageOfBooks = (null == nextPageToken ?
-                (from book in _dbcontext.Books orderby book.Id select book) :
-                (from book in _dbcontext.Books
-                 where book.Id > Int32.Parse(nextPageToken)
-                 orderby book.Id
-                 select book)).Take(pageSize + 1);
+            IQueryable<Book> query = from book in _dbcontext.Books orderby book.Id select book;

```

Comment:
A quick scan of the LINQ documentation shows tons of queries, and no
lambdas:
https://msdn.microsoft.com/en-us/library/bb397933.aspx

My book
http://www.amazon.com/Professional-ASP-NET-MVC-Jon-Galloway/dp/1118794753
shows a couple LINQ queries for reading data from the DbSet<>.  It does not
use lambdas to read data from a DbSet<>.

I expect the query syntax to be far more idiomatic.  It could only be
described as idiomatic given there is nothing like it outside C# and VB.

Shall we ask jbsimon to break the tie?

On Fri, Feb 12, 2016 at 2:21 PM, remi Taylor <notifications@github.com>
wrote:

> In aspnet/2-structured-data/Models/DbBookStore.cs
> <https://github.com/GoogleCloudPlatform/getting-started-dotnet/pull/4#discussion_r52806587>
> :
>
> > @@ -46,12 +46,13 @@ public void Delete(long id)
> >          // [START list]
> >          public BookList List(int pageSize, string nextPageToken)
> >          {
> > -            var pageOfBooks = (null == nextPageToken ?
> > -                (from book in _dbcontext.Books orderby book.Id select book) :
> > -                (from book in _dbcontext.Books
> > -                 where book.Id > Int32.Parse(nextPageToken)
> > -                 orderby book.Id
> > -                 select book)).Take(pageSize + 1);
> > +            IQueryable<Book> query = from book in _dbcontext.Books orderby book.Id select book;
>
> Hmm. Since System.Linq IEnumerable extension methods, I've favored lambda
> expressions over query syntax. I find it to be more concise and readable. I
> wonder if one is more idiomatic than the other for these use-cases? For
> this, for example, I find it *really strange* that you have to specify
> that you want to select book from books when you're *already* querying
> the Books table.
>
> That LOC repeats the fact that you want to query a Book 3 separate times? [image:
> :flushed:]
>
> Either way, the query syntax is 93 characters (vs 73). Might want to
> consider adding newlines.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/getting-started-dotnet/pull/4/files#r52806587>
> .
>